File: platform/challenges/svc4-nanofleet/service/internal/eno/eno.go
package eno

import "net/http"

type Task struct {
	Method         string `json:"method"`
	Flag           string `json:"flag"`
	CurrentRoundID int    `json:"current_round_id"`
	RelatedRoundID int    `json:"related_round_id"`
	VariantID      int    `json:"variant_id"`
}

func Authorized(r *http.Request, secret string) bool {
	sent := r.Header.Get("X-Checker-Secret")
	if sent == "" {
		sent = r.Header.Get("X-Service-Secret")
	}
	return sent != "" && sent == secret
}

File: platform/challenges/svc1-ledgerforge/service/src/eno.rs
use serde::Deserialize;

use crate::state::AppState;

#[derive(Debug, Deserialize)]
pub struct Task {
    pub method: Option<String>,
    pub flag: Option<String>,
    pub current_round_id: Option<i64>,
    pub related_round_id: Option<i64>,
    pub variant_id: Option<u8>,
    pub attack_info: Option<String>,
}

pub fn authorized(headers: &axum::http::HeaderMap, state: &AppState) -> bool {
    let sent = headers
        .get("x-checker-secret")
        .or_else(|| headers.get("x-service-secret"))
        .and_then(|v| v.to_str().ok())
        .unwrap_or("");
    !sent.is_empty() && sent == state.checker_secret
}
